examples/spark2/src/main/scala/org/apache/carbondata/examples/CompareTest.scala

```diff
@@ -0,0 +1,372 @@
+      "filter scan",
+      "filter on 2 dimensions, small result set, fetch all columns"
+    ),
+    Query(
+      "select * from $table where city='city1' and country='country2' and planet ='planet3' ",
+      "filter scan",
+      "filter on 3 dimensions, small result set, fetch all columns"
+    ),
+    Query(
+      "select * from $table where m1 < 3",
+      "filter scan",
+      "filter on measure, small result set, fetch all columns"
+    ),
+    Query(
+      "select * from $table where id like '1%' ",
+      "fuzzy filter scan",
+      "like filter, big result set"
+    ),
+    Query(
+      "select * from $table where id like '%111'",
+      "fuzzy filter scan",
+      "like filter, medium result set"
+    ),
+    Query(
+      "select * from $table where id like 'xyz%' ",
+      "fuzzy filter scan",
+      "like filter, full scan but not exist"
+    )
+  )
+
+  private def loadParquetTable(spark: SparkSession, input: DataFrame): Double = time {
+    // partitioned by last 1 digit of id column
+    val dfWithPartition = input.withColumn("partitionCol", input.col("id").%(10))
+    dfWithPartition.write
+        .partitionBy("partitionCol")
+        .mode(SaveMode.Overwrite)
+        .parquet(parquetTableName)
+  }
+
+  private def loadCarbonTable(spark: SparkSession, input: DataFrame, version: String): Double = {
+    CarbonProperties.getInstance().addProperty(
+      CarbonCommonConstants.CARBON_DATA_FILE_VERSION,
+      version
+    )
+    spark.sql(s"drop table if exists ${carbonTableName(version)}")
+    time {
+      input.write
+          .format("carbondata")
+          .option("tableName", carbonTableName(version))
+          .option("tempCSV", "false")
+          .option("single_pass", "true")
+          .option("dictionary_exclude", "id") // id is high cardinality column
+          .mode(SaveMode.Overwrite)
+          .save()
+    }
+  }
+
+  // load data into parquet, carbonV2, carbonV3
+  private def prepareTable(spark: SparkSession): Unit = {
+    val df = generateDataFrame(spark)
+    println(s"loading ${df.count} records, schema: ${df.schema}")
+    val loadParquetTime = loadParquetTable(spark, df)
+    val loadCarbonV3Time = loadCarbonTable(spark, df, version = "3")
+    println(s"load completed, time: $loadParquetTime, $loadCarbonV3Time")
+    spark.read.parquet(parquetTableName).registerTempTable(parquetTableName)
+  }
+
+  // Run all queries for the specified table
+  private def runQueries(spark: SparkSession, tableName: String): Array[(Double, Array[Row])] = {
+    println(s"start running queries for $tableName...")
+    var result: Array[Row] = null
+    queries.zipWithIndex.map { case (query, index) =>
+      val sqlText = query.sqlText.replace("$table", tableName)
+      print(s"running query ${index + 1}: $sqlText ")
+      val rt = time {
+        result = spark.sql(sqlText).collect()
+      }
+      println(s"=> $rt sec")
+      (rt, result)
```

Comment:
Don't cache the result, it is leading to memory issues if more than 2 million data. Just keep the length of the data is enough